runtime/vm/jvminit.c

```diff
@@ -1856,7 +1856,69 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
 				}
 			}
 #endif /* !defined(WIN32) && !defined(J9ZTPF) */
+			/* Parse options related to Large Pages */
+			{
+				IDATA argIndexUseLargePages = FIND_AND_CONSUME_ARG(EXACT_MATCH, MAPOPT_XXUSELARGEPAGES, NULL);
+				IDATA argIndexLargePageSizeInBytes = FIND_AND_CONSUME_ARG(STARTSWITH_MATCH, MAPOPT_XXLARGEPAGESIZEINBYTES_EQUALS, NULL);
+
+				if (argIndexUseLargePages > argIndexLargePageSizeInBytes) {
+					vm->largePageArgIndex = argIndexUseLargePages;
+					vm->largePageSizeRequested = -1;
+				}
+				else if (argIndexLargePageSizeInBytes >= 0) {
+					vm->largePageArgIndex = argIndexLargePageSizeInBytes;
+					char* memoryValue = NULL;
+					parseError = GET_OPTION_VALUE(argIndexLargePageSizeInBytes, '=', &memoryValue);
 
+					if (OPTION_OK != parseError) {
+						parseErrorOption = MAPOPT_XXLARGEPAGESIZEINBYTES_EQUALS;
+						goto _memParseError;
+					}
+
+   					UDATA qualifierShiftAmount = 0;
+					UDATA requestedLargeCodePageSize = 0;
+
+					UDATA scanResult = scan_udata(&memoryValue, &requestedLargeCodePageSize);
+
+					/* First scan for the integer string */
+					if (0 != scanResult) {
+						if (1 == scanResult)
+							j9nls_printf(PORTLIB, J9NLS_ERROR, J9NLS_VM_OPTIONS_MUST_BE_NUMBER);
+						else
+							j9nls_printf(PORTLIB, J9NLS_ERROR, J9NLS_VM_OPTION_OVERFLOW);
+						j9nls_printf(PORTLIB, J9NLS_ERROR, J9NLS_VM_OPTIONS_INCORRECT_MEMORY_SIZE, "-XX:LargePageSizeInBytes=");
+						parseErrorOption = MAPOPT_XXLARGEPAGESIZEINBYTES_EQUALS;
+						goto _memParseError;
+					}
+
+					if(try_scan(&memoryValue, "G") || try_scan(&memoryValue, "g"))
```

Comment:
There is a GET_MEM_VALUE option that does this, i.e. GET_MEMORY_VALUE() can be used instead of GET_OPTION_VALUE().

Reply:
Thanks for the constructive feedback. 
I've replaced it here, and it works as expected. 
https://github.com/eclipse/openj9/pull/7476/commits/057309cd5b9ffd2d4d06db06c183706acdc6e355

Reply:
Thanks for the constructive feedback. 
I've made that changed here, and it looks to pass all my tests. 
https://github.com/eclipse/openj9/pull/7476/commits/057309cd5b9ffd2d4d06db06c183706acdc6e355

Reply:
The code won't compile on all platforms as variables in c code need to be declared at the start of a scope. Also lpOption should be initialized to MAPOPT_XXLARGEPAGESIZEINBYTES_EQUALS.